### DISCUSSION TITLE
Migrating from flask to fastAPI : failed requests in load test

### DISCUSSION THREAD

**Discussion by Aeym** [fastapi/fastapi#6832] [Questions] [answered]
Hello everyone,

I recently tried to migrate from flask to fastAPI. Everything seemed to work just perfectly until I tried to load test my fastAPI api with [Locust](https://locust.io/). I have many failed requests in this scenario, that I didn't have before. The non-api-code hasn't change, but it seems that the asynchronous behavior end-up mixing request. In fact, endpoints are working fine but data sent to my functions are terrible, and i end up with random python errors like the following : 

`int is not supscritable`

`too many values to unpack`

etc...

I had similar results using [unittest-parallel](https://pypi.org/project/unittest-parallel/) (again, not any failed request with flask)
Note that I litterally have ZERO erros when I test it with unittest (not in parallel).

Anyone has experienced the same issue ?

Thanks a lot

**Comment by krishnardt** [comment]
Can you show an example code apart from just an issue?

that issue could happen when we have LHS(variables) < RHS(data or returning function).

**Comment by Aeym** [comment, marked as answer]
It was not a problem with fastAPI but in the way I was instancing one of my class. I was instancing my class only one time, and not everytime the endpoint was called (end up in mix class data). Thanks for the fast answer tho !

**Comment by tiangolo** [comment]
Thanks for the help here @krishnardt ! 👏  🙇 

Thanks for reporting back and closing the issue @Aeym  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
